sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -416,6 +415,97 @@ private object CosmosChangeFeedConfig {
+
+private object CosmosThroughputControlConfig {
+    private val enableThroughputControlSupplier = CosmosConfigEntry[Boolean](
+        key = "spark.cosmos.enableThroughputControl",
+        mandatory = false,
+        defaultValue = Some(false),
+        parseFromStringFunction = enableThroughputControl => enableThroughputControl.toBoolean,
+        helpMessage = "A flag to indicate whether throughput control is enabled.")
+
+    private val groupNameSupplier = CosmosConfigEntry[String](
+        key = "spark.cosmos.throughputControl.name",
+        mandatory = false,
+        parseFromStringFunction = groupName => groupName,
+        helpMessage = "Throughput control group name. " +
+            "Since customer is allowed to create many groups for a container, the name should be unique.")
+
+    private val targetThroughputSupplier = CosmosConfigEntry[Int](
+        key = "spark.cosmos.throughputControl.targetThroughput",
+        mandatory = false,
+        parseFromStringFunction = targetThroughput => targetThroughput.toInt,
+        helpMessage = "Throughput control group target throughput. The value should be larger than 0.")
+
+    private val targetThroughputThresholdSupplier = CosmosConfigEntry[Double](
+        key = "spark.cosmos.throughputControl.targetThroughputThreshold",
+        mandatory = false,
+        parseFromStringFunction = targetThroughput => targetThroughput.toDouble,
+        helpMessage = "Throughput control group target throughput threshold. The value should be between (0,1]. ")
+
+    private val globalControlDatabaseSupplier = CosmosConfigEntry[String](
+        key = "spark.cosmos.throughputControl.globalControl.database",
+        mandatory = false,
+        parseFromStringFunction = globalControlDatabase => globalControlDatabase,
+        helpMessage = "Database which will be used for throughput global control.")
+
+    private val globalControlContainerSupplier = CosmosConfigEntry[String](
+        key = "spark.cosmos.throughputControl.globalControl.container",
+        mandatory = false,
+        parseFromStringFunction = globalControlContainer => globalControlContainer,
+        helpMessage = "Container which will be used for throughput global control.")
+
+    private val globalControlItemRenewIntervalSupplier = CosmosConfigEntry[Duration](
+        key = "spark.cosmos.throughputControl.globalControl.renewIntervalInSeconds",
+        mandatory = false,
+        parseFromStringFunction = renewIntervalInSeconds => Duration.ofSeconds(renewIntervalInSeconds.toInt),
+        helpMessage = "This controls how often the client is going to update the throughput usage of itself " +
+            "and adjust its own throughput share based on the throughput usage of other clients")
+
+    private val globalControlItemExpireIntervalSupplier = CosmosConfigEntry[Duration](
+        key = "spark.cosmos.throughputControl.group.globalControl.expireIntervalInSeconds",
```

Comment:
the config name is not consistent with others. do we need the additional `group` in the config name?

shouldn't this be `spark.cosmos.throughputControl.globalControl.expireIntervalInSeconds` ?

Reply:
Good catch, thanks Mo, will change in next iteration